plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,121 @@
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_enriched_event(self, criteria: dict, exclusions: dict, time_range: dict, window: str) -> Optional[dict]:
+        if self.org_key == "":
+            raise PluginException(cause="There's no org key input.",
+                                  assistance="Please add a valid org key to the connection.")
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs",
+            json_data={
+                "criteria": criteria,
+                "exclusions": exclusions,
+                "time_range": time_range,
+                "window": window
+
+            },
+        )
 
-    def test(self):
-        host = self.host
-        token = self.token
-        connector = self.connector
-        devices = "/integrationServices/v3/device"
-        headers = {"X-Auth-Token": f"{token}/{connector}"}
-        url = host + devices
-
-        result = requests.get(url, headers=headers)
-        if result.status_code == 200:
-            return {"success": True}
-        if result.status_code == 401:
-            raise ConnectionTestException(preset=ConnectionTestException.Preset.API_KEY)
-        raise ConnectionTestException(
-            f"An unknown error occurred. Response code was: {result.status_code}"
-            f" If the problem persists please contact support for help. Response was: {result.text}"
+        return response.get("job_id")
+
+    def get_enriched_event_status(self, job_id: str = None) -> bool:
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v1/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}",
+            json_data={"cb_job_id": job_id},
+        )
+        contacted = response.get("contacted")
+        completed = response.get("completed")
+        if contacted and completed and (contacted == completed):
+            return True
+        return False
+
+    def retrieve_results_for_enriched_event(self, job_id: str = None) -> Optional[dict]:
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}/results",
+            json_data={"job_id": job_id},
         )
+
+        return response
+
+    def get_job_id_for_detail_search(self, event_ids: str) -> Optional[str]:
+        if self.org_key == "":
+            raise PluginException(cause="There's no org key input.",
+                                  assistance="Please add a valid org key to the connection.")
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/detail_jobs",
+            json_data={"event_ids": [event_ids]},
```

Comment:
this line suggests that the API is expecting a list of event_ids. Is there a reason we only pass one event_id str in? Could totally be weird API design just wondering.

Reply:
<img width="847" alt="Screen Shot 2021-12-21 at 12 34 25 PM" src="https://user-images.githubusercontent.com/87443773/146973776-33f89934-8d0e-412b-9a16-7c11a9ad33e3.png">
Screenshot of the API design

Reply:
Also the cloud console for requests implies there's one event ID per event I believe, I'll double check